pkg/collector/corechecks/snmp/config.go

```diff
@@ -0,0 +1,273 @@
+package snmp
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"gopkg.in/yaml.v2"
+	"path/filepath"
+)
+
+var defaultOidBatchSize = 60
+var defaultPort = uint16(161)
+var defaultRetries = 3
+var defaultTimeout = 2
+
+type snmpInitConfig struct {
+	Profiles      profileConfigMap `yaml:"profiles"`
+	GlobalMetrics []metricsConfig  `yaml:"global_metrics"`
+}
+
+type snmpInstanceConfig struct {
+	IPAddress        string            `yaml:"ip_address"`
+	Port             uint16            `yaml:"port"`
+	CommunityString  string            `yaml:"community_string"`
+	SnmpVersion      string            `yaml:"snmp_version"`
+	Timeout          int               `yaml:"timeout"`
+	Retries          int               `yaml:"retries"`
+	User             string            `yaml:"user"`
+	AuthProtocol     string            `yaml:"authProtocol"`
+	AuthKey          string            `yaml:"authKey"`
+	PrivProtocol     string            `yaml:"privProtocol"`
+	PrivKey          string            `yaml:"privKey"`
+	ContextName      string            `yaml:"context_name"`
+	Metrics          []metricsConfig   `yaml:"metrics"`
+	MetricTags       []metricTagConfig `yaml:"metric_tags"`
+	Profile          string            `yaml:"profile"`
+	UseGlobalMetrics bool              `yaml:"use_global_metrics"`
+}
+
+type snmpConfig struct {
```

Comment:
Are you referring to `snmpInstanceConfig` vs `snmpConfig` ?

snmpInstanceConfig is for unmarshalling and snmpConfig is for storing parsed/enriched instance config.

For a simpler integration I would use the same structure, but since snmp is quite complex, I think it's cleaner to keep things separate.

It's possible to use `snmpInstanceConfig` and modify it inplace (2) for transformations done at https://github.com/DataDog/datadog-agent/blob/2d652f60a77dee2ac9fc06194bb8189f05b406fc/pkg/collector/corechecks/snmp/config.go#L106-L211

Reply:
My point was that `snmpConfig`  duplicates all the data from both `snmpInitConfig` and `snmpInstanceConfig`.

Reply:
@gmarabout Is your point about consuming more memory than needed ? 

In this case, once the [Configure step](https://github.com/DataDog/datadog-agent/blob/84bdd7050eb1ff66dfa4f66e711a371c0d99dc6f/pkg/collector/corechecks/snmp/snmp.go#L107-L130) is over snmpInitConfig and snmpInstanceConfig should be garbage collected, the data shouldn't be stored twice.

Reply:
Agree this can help reducing duplication but I'm concerned it might lead to possible confusion and higher memory usage. 

For example `snmpInstanceConfig.Metrics` and `snmpConfig.metrics` are different, the first is a subset of the later. Storing the whole `snmpInstanceConfig` in `snmpConfig` migh lead to higher memory usage if `snmpInstanceConfig.Metrics` and the promoted `snmpInstanceConfig.allMetrics` (metrics from snmpInstanceConfig and profiles) are both kept (we can clean up `snmpInstanceConfig.Metrics`, but doesn't feel very clean way).